tests/unit/s2n_resume_test.c

```diff
@@ -163,6 +169,143 @@ int main(int argc, char **argv)
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&ticket_stuffer, sizeof(tls12_ticket) - S2N_TLS_SECRET_LEN));
+            /* The secret needs to be written to the ticket separately as it has a fixed length */
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&ticket_stuffer, test_master_secret.data, S2N_TLS_SECRET_LEN));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+
+            EXPECT_OK(s2n_client_deserialize_session_state(conn, &ticket_stuffer));
+
+            EXPECT_EQUAL(conn->actual_protocol_version, S2N_TLS12);
+            EXPECT_EQUAL(conn->secure.cipher_suite, &s2n_rsa_with_aes_128_gcm_sha256);
+
+            EXPECT_BYTEARRAY_EQUAL(test_master_secret.data, conn->secure.master_secret, S2N_TLS_SECRET_LEN);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+        }
+
+        /* Deserialized ticket sets correct PSK values for session resumption in TLS1.3 */
+        {
+            uint8_t tls13_ticket[] = {
+                S2N_TLS13_SERIALIZED_FORMAT_VERSION,
+                S2N_TLS13,
+                TLS_AES_128_GCM_SHA256,
+                TICKET_ISSUE_TIME,
+                TICKET_AGE_ADD,
+                SECRET_LEN,
+                SECRET,
+            };
+
+            struct s2n_blob ticket_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&ticket_blob, tls13_ticket, sizeof(tls13_ticket)));
+            struct s2n_stuffer ticket_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&ticket_stuffer, &ticket_blob));
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&ticket_stuffer, sizeof(tls13_ticket)));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+
+            /* Initialize client ticket */
+            uint8_t client_ticket[] = { CLIENT_TICKET };
+            EXPECT_SUCCESS(s2n_realloc(&conn->client_ticket, sizeof(client_ticket)));
+            EXPECT_MEMCPY_SUCCESS(conn->client_ticket.data, client_ticket, sizeof(client_ticket));
+
+            EXPECT_OK(s2n_client_deserialize_session_state(conn, &ticket_stuffer));
+
+            struct s2n_psk *psk = NULL;
+            EXPECT_OK(s2n_array_get(&conn->psk_params.psk_list, 0, (void**) &psk));
+            EXPECT_NOT_NULL(psk);
+
+            EXPECT_EQUAL(psk->type, S2N_PSK_TYPE_RESUMPTION);
+            S2N_BLOB_EXPECT_EQUAL(psk->identity, conn->client_ticket);
+
+            EXPECT_EQUAL(psk->secret.size, SECRET_LEN);
+            uint8_t secret[] = { SECRET };
+            EXPECT_BYTEARRAY_EQUAL(psk->secret.data, secret, sizeof(secret));
+
+            EXPECT_EQUAL(psk->hmac_alg, S2N_HMAC_SHA256);
+            
+            uint8_t ticket_age_add[] = { TICKET_AGE_ADD };
+            uint32_t expected_ticket_age_add = ticket_age_add[0] | (ticket_age_add[1] << 8) | \
+                                              (ticket_age_add[2] << 16) | (ticket_age_add[3] << 24);
```

Comment:
Since the TICKET_AGE_ADD itself is just a hardcoded series of bytes, why not also hardcode both TICKET_AGE_ADD_BYTES and TICKET_AGE_ADD? Same suggestion for issue time.

Alternatively, you could figure out the uint value of both values at the very beginning of main, since they'll never change.